src/redux/selectors/subscriptions.js

```diff
@@ -0,0 +1,278 @@
+import { SUGGESTED_FEATURED, SUGGESTED_TOP_SUBSCRIBED } from 'constants/subscriptions';
+import { createSelector } from 'reselect';
+import {
+  selectAllClaimsByChannel,
+  selectClaimsById,
+  selectAllFetchingChannelClaims,
+  makeSelectChannelForClaimUri,
+  selectClaimsByUri,
+  parseURI,
+} from 'lbry-redux';
+import { swapKeyAndValue } from 'util/swap-json';
+
+// Returns the entire subscriptions state
+const selectState = state => state.subscriptions || {};
+
+// Returns the list of channel uris a user is subscribed to
+export const selectSubscriptions = createSelector(selectState, state => state.subscriptions);
+
+// Fetching list of users subscriptions
+export const selectIsFetchingSubscriptions = createSelector(selectState, state => state.loading);
+
+// The current view mode on the subscriptions page
+export const selectViewMode = createSelector(selectState, state => state.viewMode);
+
+// Suggested subscriptions from internal apis
+export const selectSuggested = createSelector(selectState, state => state.suggested);
+export const selectIsFetchingSuggested = createSelector(
+  selectState,
+  state => state.loadingSuggested
+);
+export const selectSuggestedChannels = createSelector(
+  selectSubscriptions,
+  selectSuggested,
+  (userSubscriptions, suggested) => {
+    if (!suggested) {
+      return null;
+    }
+
+    // Swap the key/value because we will use the uri for everything, this just makes it easier
+    // suggested is returned from the api with the form:
+    // {
+    //   featured: { "Channel label": uri, ... },
+    //   top_subscribed: { "@channel": uri, ... }
+    //   top_bid: { "@channel": uri, ... }
+    // }
+    // To properly compare the suggested subscriptions from our current subscribed channels
+    // We only care about the uri, not the label
+
+    // We also only care about top_subscribed and featured
+    // top_bid could just be porn or a channel with no content
+    const topSubscribedSuggestions = swapKeyAndValue(suggested[SUGGESTED_TOP_SUBSCRIBED]);
```

Comment:
I need to take a closer look at why `swapKeyAndValue` is required in this context for sorting, I will revisit this PR

Reply:
The comment is pretty much all there is to it. Swapping the value and key so `uri` was the key made it a lot simpler (but still confusing) because I could just check if the subscription existed with 
```
alreadySubscribedToSuggestedSubscription = topSubscribedSuggestions[uri]
```
Without swapping I wasn't sure how to do that without a bunch of loops every time I needed to check.